src/google/protobuf/descriptor.cc

```diff
@@ -665,7 +665,10 @@ FileDescriptorTables::FileDescriptorTables()
 
 FileDescriptorTables::~FileDescriptorTables() {}
 
-const FileDescriptorTables FileDescriptorTables::kEmpty;
+inline const FileDescriptorTables& FileDescriptorTables::GetEmptyInstance() {
```

Comment:
Could you change it to follow the pattern we used for GetEmptyString()?
https://github.com/google/protobuf/blob/master/src/google/protobuf/generated_message_util.h#L87

AFAIK your current approach is only guaranteed thread-safe with C++11. As we haven't retired old compiler yet, the old style google_once_init approach is needed to guarantee the thread-safety.